### PR TITLE
chore(structured-properties): add cli validation for entity types

### PR DESCRIPTION
**Pull request by shirshanka**
Adds validation for entity types referenced in structured property definition yaml files.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/api/entities/structuredproperties/structuredproperties.py [hunk 1]
```diff
@@ -41,25 +41,28 @@ class AllowedValue(ConfigModel):
     description: Optional[str] = None
 
 
+VALID_ENTITY_TYPES_PREFIX_STRING = ", ".join(
+    [
+        f"urn:li:entityType:datahub.{x}"
+        for x in ["dataset", "dashboard", "dataFlow", "schemaField"]
```
Comment (david-leifker) [inline metadata-ingestion/src/datahub/api/entities/structuredproperties/structuredproperties.py:47]: This is far from a complete list and the valid types vary per structured property definition. Just a temporary or example?

Comment (david-leifker) [reply metadata-ingestion/src/datahub/api/entities/structuredproperties/structuredproperties.py:47]: Does the python code have access to the entity registry or is that server side only?

Comment (david-leifker) [reply metadata-ingestion/src/datahub/api/entities/structuredproperties/structuredproperties.py:47]: Hmm, I wonder if you might parse this `http://localhost:9002/openapi/v3/api-docs/10-openapi-v3` as a workaround.

Comment (david-leifker) [reply metadata-ingestion/src/datahub/api/entities/structuredproperties/structuredproperties.py:47]: That's the openapi spec that is generated from the entity registry on the running server as a proxy for the entity registry's data.

Comment (shirshanka) [reply metadata-ingestion/src/datahub/api/entities/structuredproperties/structuredproperties.py:47]: the problem is that technically - one could register a new entityType completely dynamically. So generating a valid list of entityTypes by consulting the server requires talking to the data plane (not just the entity registry / openapi spec)

Comment (shirshanka) [reply metadata-ingestion/src/datahub/api/entities/structuredproperties/structuredproperties.py:47]: This list of entities is just a sample... the error message looks like this:
```
Input urn:li:entityType:dataset is not a valid entity type urn. Valid entity type urns are urn:li:entityType:datahub.dataset, urn:li:entityType:datahub.dashboard, urn:li:entityType:datahub.dataFlow, urn:li:entityType:datahub.schemaField, etc... Ensure that the entity type is valid. (type=value_error)
```

**Review by david-leifker — APPROVED** [review]

**Pull request merged**